src/Kerbero.Domain/NukiActions/Repositories/INukiSmartLockPersistentRepository.cs

```diff
@@ -6,4 +6,5 @@ namespace Kerbero.Domain.NukiActions.Repositories;
 public interface INukiSmartLockPersistentRepository
 {
     Task<Result<NukiSmartLock>> CreateSmartLock(NukiSmartLock nukiSmartLock);
```

Comment:
Create should be enough. The scope is already defined by the Repository (SmartLock-REPOSITORY). It is tacit that you are creating a "SmartLock" and it will never create a different Entity. 

The method Read can be renamed to GetById, but it's mandatory.
IMHO, methods for reading record from the DB should be in the form:
GetById, GetByFirstName, GetAll, GetAllInRange and so on and so forth.